### PR TITLE
Use json-big to parse main query response

### PR DESCRIPTION
**Pull request by albireox**
Related to sdss/valis#23.

This PR modifies how the response JSON from the API query for the main search page is parsed. Normally big integers are not preserved when parsin JSON; instead they are cast to float and then back to integer, which causes rounding errors. Using `json-bigint` solves this issue.

See [here](https://stackoverflow.com/questions/43787712/axios-how-to-deal-with-big-integers) for more details. I haven't confirmed if there are other places in which this transformation need to be done.

### CODE REVIEW

File: src/views/Search.vue [hunk 1]
```diff
@@ -164,23 +166,29 @@ async function submit_form(this: any) {
     console.log('submitting', formData.value)
 
     // submit the POST request to Valis
+    axios.interceptors.request.use((request) => {
+      request.transformResponse = [data => data]
+      return request
+    })
+
```
Comment (havok2063) [inline src/views/Search.vue:173]: I do something similar in https://github.com/sdss/zora/blob/main/src/views/Target.vue#L176-L187

File: src/views/Search.vue [hunk 2]
```diff
@@ -86,11 +86,13 @@ import TextInput from '@/components/TextInput.vue'
 import DropdownSelect from '@/components/DropdownSelect.vue';
 import { useRouter } from 'vue-router';
 import { useAppStore } from '@/store/app'
+import JSONbigint from 'json-bigint'
```
Comment (havok2063) [inline src/views/Search.vue:89]: I use something similar in https://github.com/sdss/zora/blob/a925ff46da5b434f160df2376d2aa6e20eac253c/src/views/Target.vue#L103

**Review by havok2063 — CHANGES_REQUESTED** [review]
So it turns out I encountered the exact same issue, and solved in the [Target.vue ](https://github.com/sdss/zora/blob/main/src/views/Target.vue) code.  I used a different package, [json-big](https://www.npmjs.com/package/json-big), but it looks the same as [json-bigint](https://www.npmjs.com/package/json-bigint).  In fact `json-big` appears to a fork of `json-bigint` and I don't see any major differences between the two.  I would suggest we consolidate to a single package.  I'm happy to switch to `json-bigint`, as that seems to be the more updated and main package.  

It would also be good if we could apply the `axios` tweak to transformResponse globally for all uses.  I'm not sure what the best way is to do that though.  If that's not straightforward then we can leave the separate tweaks for now.

File: src/views/Search.vue (hunk 1 above)
Comment (albireox) [reply src/views/Search.vue]: Well, you'd think I would have read a bit more into the code 🥲

File: src/views/Search.vue (hunk 2 above)
Comment (albireox) [reply src/views/Search.vue]: I'm happy to keep `json-big`. They seem to do the same thing.

**Comment by albireox** [comment]
I've kept `json-big` since I don't think there are big (pun intended) differences.

Have a look at the new commits I made. I've moved the configuration of the axios instance to an `api.ts` file (the name and location can change) where the JSON bigint handling is done. Then you can use that instance to do API requests and it's not necessary to do the same bigint configuration each time. I used a 5 second timeout which may be insufficient in some cases.

I made this change for `Target` and `Search` but there may be other places that need it. I did a bit of testing and it seems to work but it would be good if you can test independently. The docs for the axios instance are here

https://axios-http.com/docs/instance

File: src/api.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import axios from 'axios'
+import JSONbig from 'json-big'
+
+// Defines the API instance in Axios with special handling for big integers.
+export const apiInstance = axios.create({
```
Comment (havok2063) [inline src/api.ts:5]: Can we rename the instance to `axiosInstance`?  if we rename this file to `axios.ts`, will that cause any import conflicts with the main `axios` package.

File: src/views/Target.vue [hunk 4]
```diff
@@ -225,7 +210,7 @@ async function get_db_info() {
         return
     }
 
-    await axios.get(import.meta.env.VITE_API_URL + '/info/database')
+    await apiInstance.get(import.meta.env.VITE_API_URL + '/info/database')
```
Comment (havok2063) [inline src/views/Target.vue:213]: I guess we can now remove the base url here?

File: src/views/Target.vue [hunk 5]
```diff
@@ -171,23 +171,8 @@ async function get_target_info() {
         import.meta.env.VITE_API_URL + `/target/pipelines/${sdss_id}?release=${rel}`
```
Comment (havok2063) [inline src/views/Target.vue:171]: will this still work if we remove the base urls here?

**Review by havok2063 — COMMENTED** [review]
Your changes seem reasonable to me.  I made a few comments.  We are using `axios` in a bunch of places.  Should we replace them everywhere?  The only places where it might make sense, but really probably doesn't matter, are the `ConeSearch` and `TargetResolver` components.  But at least we wouldn't need to always specify the base url.  

Dare I suggest you try to write some unit tests for these changes?  Feel free to say no.  I haven't been consistent about it but I occasionally write some.  You can take a look at the `components/__test__` directory.

**Comment by havok2063** [comment]
Actually, to keep this PR focused on the main query, we don't need to replace all instances of `axios`.  We can do that in a later PR.  I'll create a new issue for it.  I'll need to add global custom headers for the authentication to this config anyways

File: src/api.ts (hunk 3 above)
Comment (albireox) [reply src/api.ts]: Done

File: src/views/Target.vue (hunk 4 above)
Comment (albireox) [reply src/views/Target.vue]: Done

File: src/views/Target.vue (hunk 5 above)
Comment (albireox) [reply src/views/Target.vue]: Yes, it seems the URLs on `axiosInstance` need to have the initial / but they are concatenated with the base URL. It also accepts absolute URLs and the configuration (the `json-big` parsing) still gets applied.

**Comment by albireox** [comment]
I agree, let's do that as a different PR, but I'll add myself to #25. And I'll write some tests; I'm actually interested in learning how to test JS and I see you have some examples there.

File: src/api.ts (hunk 3 above)
Comment (albireox) [reply src/api.ts]: And yes, as long as the import is prefixed by `@/` it's a relative import and it can have the same names as the axios package.

**Review by havok2063 — APPROVED** [review]

**Comment by havok2063** [comment]
Looks good to me!

**Pull request merged**